torch/csrc/api/include/torch/data/datasets/shared.h

```diff
@@ -62,6 +62,11 @@ class SharedBatchDataset : public BatchDataset<
     return dataset_.get();
   }
 
+  /// Calls `reset()` on the underlying dataset.
+  void reset() {
```

Comment:
This is confusing, because now you can't make a shared wrapper around a dataset that doesn't have `reset()`? Why?

Reply:
Wow I had no idea that's the case. C++ is such a messed up language.